deposits.md

```diff
@@ -0,0 +1,225 @@
+    bytes32 hash
+  ) external;
+
+  function l1Number() view;
+  function l1Timestamp() view;
+  function l1BaseFee() view;
+  function l1Hash() view;
+}
+```
+
+## L1 Transaction Deposits
+
+[l1-transaction-deposits]: #l1-transaction-deposits
+
+L1 Transaction Deposits are [Deposit Transactions][deposit-transaction-type] generated by the
+[L2 Chain Derivation][derivation] process. The values of each transaction are determined by the
+corresponding `TransactionDeposited` event emitted by the [Deposit Feed
+contract][deposit-feed-contract] on L1.
+
+1. `from` is unchanged from the emitted value (though it may have been transformed to an alias in
+   the Deposit Feed contract).
+2. `to` may be either:
+    1. any 20-byte address (including the zero-address)
+    2. `null` in which case a contract is created.
+3. `value` is unchanged from the emitted value.
+4. `gaslimit` is unchanged from the emitted value.
+5. `data` is unchanged from the emitted value. Depending on the value of `to` it is handled as
+   either calldata or initialization code depending on the value of `to`.
+
+### Deposit Feed Contract
+
+[deposit-feed-contract]: #deposit-feed-contract
+
+The Deposit Feed contract is deployed to L1. Deposited Transactions are derived from the values in
+the `TransactionDeposited` event(s) emitted by the Deposit Feed contract.
+
+The Deposit Feed handles two special cases:
+
+1. A contract creation deposit, which is indicated by setting the `isCreation` flag to `true`.
+   In the event that the `to` address is non-zero, the contract will revert.
+2. A call from a contract account, in which case the `from` value is transformed to its L2 alias (by
+   adding `0x1111000000000000000000000000000000001111`). This prevents attacks in which a contract
+   on L1 has the same address as a contract on L2 but doesn't have the same code. We can safely
+   ignore this for EOAs because they're guaranteed to have the same "code" (i.e. no code at all).
+   This also makes it possible for users to interact with contracts on L2 even when the Sequencer is
+   down.
+
+> **TODO** Define if/how ETH withdrawals occur.
+
+A solidity-like pseudocode implementation demonstrates the functionality:
+
+```solidity
+contract DepositFeed {
+
+  event TransactionDeposited(
+    address indexed from,
+    address indexed to,
+    uint256 value,
+    uint256 gasLimit,
+    bool isCreation,
+    bytes _data
+  );
+
+  function depositTransaction(
+    address to,
+    uint256 value,
+    uint256 gasLimit,
+    bool isCreation,
+    bytes memory _data
+  ) external payable {
+    address from;
+    if (msg.sender == tx.origin) {
+        from = msg.sender;
+    } else {
+        from = msg.sender + 0x1111000000000000000000000000000000001111;
+    }
+
+    if(isCreation && _to != address(0)) {
+        revert('Contract creation deposits must not specify a recipient address.');
+    } else {
+      emit TransactionDeposited(
+        msg.sender,
+        to,
+        msg.value,
```

Comment:
This is missing `gasLimit`

Reply:
It was brought up earlier, first we concluded it may be better without to keep scope small of deposit rollup, but it makes sense to have indeed.

Reply:
It's listed in the transaction type spec, so it needs to be here for consistency. Also without it, it's possible to generate infinite computation for the rollup, which is not acceptable even for the deposit rollup imho.